src/runtime/propensity-server-test.js

```diff
@@ -296,4 +312,132 @@ describes.realWin('PropensityServer', {}, env => {
+          receivedContext = context;
+        });
+    propensityServer.sendClientEvent_(defaultEvent);
+    expect(receivedType).to.equal(PropensityApi.Event.IMPRESSION_OFFERS);
+    const str = JSON.stringify(defaultEvent.additionalParameters);
+    expect(receivedContext).to.equal(str);
+
+    //now ensure it properly converts each analytics event it receives into
+    //an appropriate propensity event (or suppresses the event entirely if it)
+    //isn't a propensity event.
+    for (const k in AnalyticsEvent) {
+      defaultEvent.eventType = AnalyticsEvent[k];
+      receivedType = null;
+      propensityServer.sendClientEvent_(defaultEvent);
+      switch (defaultEvent.eventType) {
+        case AnalyticsEvent.UNKNOWN:
+          expect(receivedType).to.equal(null);
+          break;
+        case AnalyticsEvent.IMPRESSION_PAYWALL:
+          expect(receivedType).to.equal(PropensityApi.Event.IMPRESSION_PAYWALL);
+          break;
+        case AnalyticsEvent.IMPRESSION_AD:
+          expect(receivedType).to.equal(PropensityApi.Event.IMPRESSION_AD);
+          break;
+        case AnalyticsEvent.IMPRESSION_OFFERS:
+          expect(receivedType).to.equal(PropensityApi.Event.IMPRESSION_OFFERS);
+          break;
+        case AnalyticsEvent.IMPRESSION_SUBSCRIBE_BUTTON:
+          expect(receivedType).to.equal(null);
+          break;
+        case AnalyticsEvent.IMPRESSION_SMARTBOX:
+          expect(receivedType).to.equal(null);
+          break;
+        case AnalyticsEvent.ACTION_SUBSCRIBE:
+          expect(receivedType).to.equal(null);
+          break;
+        case AnalyticsEvent.ACTION_PAYMENT_COMPLETE:
+          expect(receivedType).to
+              .equal(PropensityApi.Event.ACTION_PAYMENT_COMPLETED);
+          break;
+        case AnalyticsEvent.ACTION_ACCOUNT_CREATED:
+          expect(receivedType).to.equal(null);
+          break;
+        case AnalyticsEvent.ACTION_ACCOUNT_ACKNOWLEDGED:
+          expect(receivedType).to.equal(null);
+          break;
+        case AnalyticsEvent.ACTION_SUBSCRIPTIONS_LANDING_PAGE:
+          expect(receivedType).to
+              .equal(PropensityApi.Event.ACTION_SUBSCRIPTIONS_LANDING_PAGE);
+          break;
+        case AnalyticsEvent.ACTION_PAYMENT_FLOW_STARTED:
+          expect(receivedType).to
+              .equal(PropensityApi.Event.ACTION_PAYMENT_FLOW_STARTED);
+          break;
+        case AnalyticsEvent.ACTION_OFFER_SELECTED:
+          expect(receivedType).to
+              .equal(PropensityApi.Event.ACTION_OFFER_SELECTED);
+          break;
+        case AnalyticsEvent.EVENT_PAYMENT_FAILED:
+          expect(receivedType).to.equal(null);
+          break;
+        case AnalyticsEvent.EVENT_CUSTOM:
+          expect(receivedType).to.equal(PropensityApi.Event.EVENT_CUSTOM);
+          break;
+        default:
+          expect(false).to.be.true; //add new analytics event to the switch
+      }
+    }
+  });
+
+  it('should respect the SwG logging experiment flag', () => {
```

Comment:
Better to break this up into two tests.
// Mock the EventManager.registerEventListener() method and capture the callback

     numFetchApiCalls = 0;
    sandbox.stub(Xhr.prototype, 'fetch',
        (url, init) => {
          numFetchApiCalls++;
          return Promise.reject(new Error('Publisher not whitelisted'));
        });

One where it 'should send SwG events to Propensity Service'
// Enable experiment flag
// invoke the callback and catch
captureCallback(swgOriginatedEvent).then(() => {
  throw new Error('must have failed');
}).catch(reason => {
  expect(numFetchApiCalls).to.equal(0);
  expect(() => throw reason).to.throw(/Server down/);
});

One where it 'should not send SwG events to Propensity Service'
// same setup
captureCallback(swgOriginatedEvent).then(() => {
  throw new Error('must have failed');
}).catch(reason => {
  expect(numFetchApiCalls).to.equal(0);
  expect(() => throw reason).to.throw(/Server down/);
});

Reply:
Ok